client/dashboard/sites/settings-mcp/index.tsx

```diff
@@ -0,0 +1,290 @@
+					> = {};
+					Object.entries( siteSettings?.mcp_settings?.mcp_abilities || {} ).forEach(
+						( [ abilityId, ability ] ) => {
+							autoEnabledAbilities[ abilityId ] = {
+								...ability,
+								enabled: true, // Auto-enable all abilities
+							};
+						}
+					);
+					return {
+						...prev,
+						mcp_enabled: enabled,
+						mcp_abilities: autoEnabledAbilities,
+					};
+				}
+				// When disabling MCP, disable all abilities
+				return {
+					...prev,
+					mcp_enabled: enabled,
+					mcp_abilities: {},
+				};
+			} );
+		};
+
+		const handleAbilityChange = ( abilityId: string, enabled: boolean ) => {
+			setFormData( ( prev ) => ( {
+				...prev,
+				mcp_abilities: {
+					...prev.mcp_abilities,
+					[ abilityId ]: {
+						...prev.mcp_abilities[ abilityId ],
+						enabled,
+					},
+				},
+			} ) );
+		};
+
+		// Get abilities from the site settings data, but use form data for current state
+		const abilities = availableAbilities.map( ( [ abilityId, ability ] ) => [
+			abilityId,
+			{
+				...ability,
+				enabled: formData.mcp_enabled
+					? formData.mcp_abilities?.[ abilityId ]?.enabled ?? ability.enabled
+					: false,
+			},
+		] );
+
+		return (
+			<form onSubmit={ handleSubmit }>
+				<VStack spacing={ 6 }>
+					{ /* Main MCP toggle in its own card - only show if abilities are available */ }
+					{ hasAbilities && (
+						<Card>
+							<CardBody>
+								<CheckboxControl
+									__nextHasNoMarginBottom
+									label={ __( 'Enable Model Context Protocol (MCP)' ) }
+									help={ __(
+										'Allow AI assistants to access your site data through the Model Context Protocol.'
+									) }
+									checked={ formData.mcp_enabled }
+									onChange={ handleMcpEnabledChange }
+								/>
+							</CardBody>
+						</Card>
+					) }
+
+					{ /* Abilities settings in a separate card - always show if abilities exist */ }
+					{ hasAbilities && (
+						<Card>
+							<CardBody>
+								<VStack spacing={ 4 }>
+									<h3 style={ { margin: 0, fontSize: '16px', fontWeight: 600 } }>
+										{ __( 'MCP Abilities' ) }
+									</h3>
+
+									{ ! formData.mcp_enabled && (
+										<p style={ { color: '#646970', fontSize: '14px', margin: 0 } }>
+											{ __( 'Enable MCP above to configure these abilities.' ) }
+										</p>
+									) }
+
+									<VStack spacing={ 3 }>
+										{ abilities.map( ( [ abilityId, ability ] ) => (
+											<CheckboxControl
+												key={ abilityId }
+												__nextHasNoMarginBottom
+												label={ ability.label }
+												help={ ability.description }
+												checked={ ability.enabled }
+												onChange={ ( checked ) => handleAbilityChange( abilityId, checked ) }
+												disabled={ ! formData.mcp_enabled }
+											/>
+										) ) }
+									</VStack>
+								</VStack>
+							</CardBody>
+						</Card>
+					) }
+
+					{ /* Show message when no abilities are available */ }
+					{ ! hasAbilities && (
```

Comment:
It would be better to return early when no abilities are available like we did for the loading state. Otherwise, it's weird there is a save button there.